sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -322,6 +333,9 @@ class Dataset[T] private[sql](
   }
 
   override def toString: String = {
+    if (name != null) {
+      return name
+    }
```

Comment:
this used to show the schema which I think is useful - is this what get called on the storage tab? how about returning both name and schema?